prime-router/build.gradle.kts

```diff
@@ -502,16 +516,29 @@ tasks.azureFunctionsPackage {
 
 tasks.register("generateVersionFile") {
     doLast {
-        val stdout = ByteArrayOutputStream()
-        exec {
-            commandLine("git", "rev-parse", "--short", "HEAD")
-            standardOutput = stdout
-        }
-        val currentCommit = stdout.toString(StandardCharsets.UTF_8).trim()
-        File(buildDir, "$azureFunctionsDir/$azureAppName/version.json").writeText("{\"commitId\": \"$currentCommit\"}")
+        file("$buildDir/$azureFunctionsDir/$azureAppName/version.json").writeText("{\"commitId\": \"$commitId\"}")
     }
 }
 
+tasks.register("generateVersionObject") {
+    val sourceDir = file("$buildDir/generated-src/version")
+    val sourceFile = file("$sourceDir/Version.kt")
+    sourceDir.mkdirs()
+    sourceFile.writeText(
+        """
+        package gov.cdc.prime.router.version
+        
+        /**
+         * Supplies information for the current build.
+         * This file is generated via Gradle task prior to compile time and should always contain the current commit hash.
+         */
+        object Version {
+          const val commitId = "$commitId"
+        }
+    """.trimIndent()
```

Comment:
Seems a tad sketchy to be writing out Kotlin here.

Any thoughts on making the code static and read from the version.json file that is output? Or perhaps read directly from `.git/HEAD` and use that info to get the hash from `.git/refs` ?

Reply:
This approach was suggested by @mkalish and is not dissimilar to the way jOOQ generates objects representing the database. I'm also not entirely certain the application can read files from those locations - the pivot to this approach was done amid difficulty attempting to implement exactly that.